pkg/aggregator/mutating_walker_test.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregator
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-openapi/spec"
+	fuzz "github.com/google/gofuzz"
+	"k8s.io/kube-openapi/pkg/util/sets"
+)
+
+func fuzzFuncs(f *fuzz.Fuzzer, refFunc func(ref *spec.Ref, c fuzz.Continue, visible bool)) {
+	invisible := 0 // == 0 means visible, > 0 means invisible
+	f.Funcs(
+		func(ref *spec.Ref, c fuzz.Continue) {
+			refFunc(ref, c, invisible == 0)
+		},
+		func(sa *spec.SchemaOrStringArray, c fuzz.Continue) {
+			*sa = spec.SchemaOrStringArray{}
+			if c.RandBool() {
+				c.Fuzz(&sa.Schema)
+			} else {
+				c.Fuzz(&sa.Property)
+			}
+			if sa.Schema == nil && len(sa.Property) == 0 {
+				*sa = spec.SchemaOrStringArray{Schema: &spec.Schema{}}
+			}
+		},
+		func(url *spec.SchemaURL, c fuzz.Continue) {
+			*url = spec.SchemaURL("http://url")
+		},
+		func(s *spec.Swagger, c fuzz.Continue) {
+			invisible++
+			c.FuzzNoCustom(s)
+			invisible--
+
+			// only fuzz those fields we walk into (we skip some like extensions)
+			c.Fuzz(&s.Parameters)
+			c.Fuzz(&s.Responses)
+			c.Fuzz(&s.Definitions)
+			c.Fuzz(&s.Paths)
+		},
+		func(p *spec.Parameter, c fuzz.Continue) {
+			invisible++
+			c.FuzzNoCustom(p)
+			invisible--
+
+			// only fuzz those fields we walk into
+			c.Fuzz(&p.Ref)
+			c.Fuzz(&p.Schema)
+			if c.RandBool() {
+				p.Items = &spec.Items{}
+				c.Fuzz(&p.Items.Ref)
+			} else {
+				p.Items = nil
+			}
+		},
+		func(s *spec.Response, c fuzz.Continue) {
+			c.FuzzNoCustom(s)
```

Comment:
this has the potential to fuzz headers visibly

Reply:
Isn't that overridden by the following call?